### PR TITLE
159799206 ensure repo async

### PR DESCRIPTION
**Pull request by bjkirk94**
Made any repository methods we could async.  Some are still improperly using IQueriables, but will hold off on fixing this outside of repos until queues are in place.

### CODE REVIEW

File: src/Housing.Selection.Context/DataAccess/AddressRepository.cs [hunk 1]
```diff
@@ -18,14 +18,14 @@ public AddressRepository(IDbContext housingSelectionContext)
         public void AddAddress(Address address)
         {
             _housingSelectionDbContext.Addresses.Add(address);
-            _housingSelectionDbContext.SaveChanges();
+            _housingSelectionDbContext.SaveChangesAsync();
```
Comment (escalonn) [inline src/Housing.Selection.Context/DataAccess/AddressRepository.cs]: must await this call!

File: src/Housing.Selection.Context/DataAccess/BatchRepository.cs [hunk 2]
```diff
@@ -18,10 +19,10 @@ public BatchRepository(IDbContext housingSelectionContext)
         public void AddBatch(Batch batch)
         {
             _housingSelectionDbContext.Batches.Add(batch);
-            _housingSelectionDbContext.SaveChanges();
+            _housingSelectionDbContext.SaveChangesAsync();
```
Comment (escalonn) [inline src/Housing.Selection.Context/DataAccess/BatchRepository.cs]: await!

File: src/Housing.Selection.Context/DataAccess/RoomRepository.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@ public RoomRepository(IDbContext housingSelectionContext)
         public void AddRoom(Room room)
         {
             _housingSelectionDbContext.Rooms.Add(room);
-            _housingSelectionDbContext.SaveChanges();
+            _housingSelectionDbContext.SaveChangesAsync();
```
Comment (escalonn) [inline src/Housing.Selection.Context/DataAccess/RoomRepository.cs]: await!

File: src/Housing.Selection.Context/DataAccess/UserRepository.cs [hunk 4]
```diff
@@ -18,7 +18,7 @@ public UserRepository(IDbContext housingSelectionContext)
         public void AddUser(User user)
         {
             _housingSelectionDbContext.Users.Add(user);
-            _housingSelectionDbContext.SaveChanges();
+            _housingSelectionDbContext.SaveChangesAsync();
```
Comment (escalonn) [inline src/Housing.Selection.Context/DataAccess/UserRepository.cs]: await

**Comment by bjkirk94** [comment]
Performed all requested changes as well as a few others that were previously missed that also meant the methods were not being used fully asynchronously.

**Comment by bjkirk94** [comment]
Note that the repositories are still not using IQueriables properly.  This is documented and will be addressed in https://www.pivotaltracker.com/story/show/159864144

**Comment by bjkirk94** [comment]
Note that the repositories are still not using IQueriables properly.  This is documented and will be addressed in https://www.pivotaltracker.com/story/show/159864144

File: src/Housing.Selection.Testing/Context/DataAccess/TestUserRepository.cs [hunk 5]
```diff
@@ -51,7 +52,7 @@ public void CanGetUsers()
         }
 
         [Fact]
-        public void CanAddUser()
+        public async void CanAddUser()
```
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestUserRepository.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7ISiZQY_aXE9UnX&open=AWVYh7ISiZQY_aXE9UnX)

[ISSUE_KEY]: AWVYh7ISiZQY_aXE9UnX

File: src/Housing.Selection.Testing/Context/DataAccess/TestBatchRepository.cs [hunk 6]
```diff
@@ -43,7 +43,7 @@ public TestBatchRepository()
         }
 
         [Fact]
-        public void CanAddBatch()
+        public async void CanAddBatch()
```
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestBatchRepository.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7HviZQY_aXE9UnR&open=AWVYh7HviZQY_aXE9UnR)

[ISSUE_KEY]: AWVYh7HviZQY_aXE9UnR

File: src/Housing.Selection.Testing/Context/DataAccess/TestUserRepository.cs (hunk 5 above)
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestUserRepository.cs]: Code Smell Code Smell: Make this test method non-'async' or return 'Task'. ([csharpsquid:S3433](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3433))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7ISiZQY_aXE9UnW&open=AWVYh7ISiZQY_aXE9UnW)

[ISSUE_KEY]: AWVYh7ISiZQY_aXE9UnW

File: src/Housing.Selection.Testing/Context/DataAccess/TestRoomRepository.cs [hunk 7]
```diff
@@ -67,7 +67,7 @@ public async void CanSaveChangesAsync()
         }
 
         [Fact]
-        public void CanAddRoom()
+        public async void CanAddRoom()
```
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestRoomRepository.cs]: Code Smell Code Smell: Make this test method non-'async' or return 'Task'. ([csharpsquid:S3433](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3433))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7IGiZQY_aXE9UnU&open=AWVYh7IGiZQY_aXE9UnU)

[ISSUE_KEY]: AWVYh7IGiZQY_aXE9UnU

Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestRoomRepository.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7IGiZQY_aXE9UnV&open=AWVYh7IGiZQY_aXE9UnV)

[ISSUE_KEY]: AWVYh7IGiZQY_aXE9UnV

File: src/Housing.Selection.Testing/Context/DataAccess/TestAddressRepository.cs [hunk 8]
```diff
@@ -42,7 +42,7 @@ public TestAddressRepository()
         }
 
         [Fact]
-        public void CanAddAddress()
+        public async void CanAddAddress()
```
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestAddressRepository.cs]: Code Smell Code Smell: Make this test method non-'async' or return 'Task'. ([csharpsquid:S3433](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3433))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7HiiZQY_aXE9UnO&open=AWVYh7HiiZQY_aXE9UnO)

[ISSUE_KEY]: AWVYh7HiiZQY_aXE9UnO

File: src/Housing.Selection.Testing/Context/DataAccess/TestNameRepository.cs [hunk 9]
```diff
@@ -42,7 +42,7 @@ public TestNameRepository()
         }
 
         [Fact]
-        public void CanAddName()
+        public async void CanAddName()
```
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestNameRepository.cs]: Code Smell Code Smell: Make this test method non-'async' or return 'Task'. ([csharpsquid:S3433](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3433))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7H7iZQY_aXE9UnS&open=AWVYh7H7iZQY_aXE9UnS)

[ISSUE_KEY]: AWVYh7H7iZQY_aXE9UnS

Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestNameRepository.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7H7iZQY_aXE9UnT&open=AWVYh7H7iZQY_aXE9UnT)

[ISSUE_KEY]: AWVYh7H7iZQY_aXE9UnT

File: src/Housing.Selection.Testing/Context/DataAccess/TestBatchRepository.cs (hunk 6 above)
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestBatchRepository.cs]: Code Smell Code Smell: Make this test method non-'async' or return 'Task'. ([csharpsquid:S3433](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3433))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7HviZQY_aXE9UnQ&open=AWVYh7HviZQY_aXE9UnQ)

[ISSUE_KEY]: AWVYh7HviZQY_aXE9UnQ

File: src/Housing.Selection.Testing/Context/DataAccess/TestAddressRepository.cs (hunk 8 above)
Comment (fredbelotte) [inline src/Housing.Selection.Testing/Context/DataAccess/TestAddressRepository.cs]: Bug Bug: Return 'Task' instead. ([csharpsquid:S3168](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS3168))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=housing%3Aselection&pullRequest=27&issues=AWVYh7HiiZQY_aXE9UnP&open=AWVYh7HiiZQY_aXE9UnP)

[ISSUE_KEY]: AWVYh7HiiZQY_aXE9UnP

**Pull request merged**